### PR TITLE
Gunsmith quest objectives

### PR DESCRIPTION
**Pull request by Ygnas**
Did the first 8 quests. Hope to finish rest of them tomorrow.
@thaddeus tell me, if you want something changed.
Gunsmith - Part 7  is missing part id for "Magpul MOE M870 forestock" or I cant find it.
#122 #2

### CODE REVIEW

File: quests.json [hunk 1]
```diff
@@ -3887,7 +3887,44 @@
...
         ],
-        "objectives": [],
+        "objectives": [
+            {
+                "type": "build",
+                "target": "54491c4f4bdc2db1078b4568",
+                "number": 1,
+                "with": [
+                    {
+                        "type": "stat",
```
Comment (thaddeus) [inline quests.json:3897]: Format for this will be really easy to use, thanks! 👍

File: quests.json [hunk 2]
```diff
@@ -4055,7 +4301,44 @@
...
+                        "name": "Tactical device"
+                    },
+                    {
+                        "type": "attachment",
+                        "name": "Foregrip"
+                    },
+                    {
+                        "type": "attachment",
+                        "name": "Magpul MOE M870 forestock",
+                        "id": ""
```
Comment (thaddeus) [inline quests.json:4336]:
```suggestion
                        "type": "part",
                        "id": "5a788068c5856700137e4c8f"
```

I think having a type with a specific `part` indication versus `attachment` which is a general category of attachment would help make the information easier to parse. And the name would be inferred from the `items.{locale}.json` file

File: quests.json [hunk 3]
```diff
@@ -4083,7 +4366,51 @@
...
+                    },
+                    {
+                        "type": "attachment",
+                        "name": "Fortis Shift tactical grip",
+                        "id": "59f8a37386f7747af3328f06"
+                    },
+                    {
+                        "type": "attachment",
+                        "name": "AK 7.62x39 Magpul PMAG 30 GEN M3 30-round magazine",
+                        "id": "59d6272486f77466146386ff"
```
Comment (thaddeus) [inline quests.json:4407]:
```suggestion
                        "type": "part",
                        "id": "59f8a37386f7747af3328f06"
                    },
                    {
                        "type": "part",
                        "id": "59d6272486f77466146386ff"
```

Same thoughts here about differentiating attachment categories and specific parts

**Review by thaddeus — COMMENTED** [review]
Couple minor suggestions, but looks great to me so far! Thanks so much!

**Comment by Ygnas** [comment]
Finished all the quests.
Few things Gunsmith - Part 9 require Zenit parts, so I did it by wiki not sure if there is few versions of them and if it should be done differently.

Gunsmith - Part 16 requires KAC QDSS NT-4 FDE 5.56x45 silencer there is 2 versions of that, so check that if need changes.

**Review by thaddeus — APPROVED** [review]
LGTM 👍 Thanks for figuring this all out! Will hopefully get it added and displayed on the tracker in a day or two!

**Pull request merged**